src/bot/commands/check-score.command.ts

```diff
@@ -83,6 +83,13 @@ export class CheckScoreCommand implements DiscordTransformedCommand<ScoreDTO> {
       interaction.guild.id,
     ]);
 
+    // Role Changes when scores are updated.
+    const top_virgins = await this.virgins.find(
+      { guild: interaction.guild },
+      { orderBy: [{ cached_dur_in_vc: -1 }], limit: 1 },
+    );
+    this.discord_helper.assignBiggestVirginRole(top_virgins[0]);
```

Comment:
It would probably be worth refactoring this into a method for reuse.

Reply:
Code has been refactored